textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/planstore/PlanStore.java

```diff
@@ -234,61 +241,54 @@ private void updatePlanInternal(String planName, String description, LogicalPlan
+     * @return String
      * @throws TextDBException
      */
-    public LogicalPlan readPlanObject(String filePath) throws TextDBException {
-        FileInputStream fis = null;
-        ObjectInputStream ois = null;
+    public String readPlanJson(String filePath) throws TextDBException {
         try {
-            fis = new FileInputStream(filePath);
-            ois = new ObjectInputStream(fis);
-            return (LogicalPlan) ois.readObject();
-        } catch (Exception e) {
-            throw new TextDBException("failed to read plan object", e);
-        } finally {
-            try {
-                ois.close();
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
+            // Reading the Logical JSON string as bytes
+            byte[] encoded = Files.readAllBytes(Paths.get(filePath));
+            return new String(encoded);
+        }
+        catch (IOException e) {
+            throw new TextDBException("Failed to read the Logical Plan's JSON");
         }
     }
 
     /**
-     * Writes the given plan object into the given file path.
-     *
-     * @param plan, the plan object to store.
-     * @param filePath, the file path of the plan object.
+     * Writes the given Logical Plan JSON to a .json file, at the given file path
+     * @param logicalPlanJson, the logical plan JSON in String format
+     * @param filePath, the file path of the logical plan JSON file
      * @throws TextDBException
      */
-    private void writePlanObject(LogicalPlan plan, String filePath) throws TextDBException {
-        FileOutputStream fos = null;
-        ObjectOutputStream oos = null;
+    private void writePlanJson(String logicalPlanJson, String filePath) throws TextDBException {
         try {
-            fos = new FileOutputStream(filePath);
-            oos = new ObjectOutputStream(fos);
-            oos.writeObject(plan);
-        } catch (IOException e) {
-            throw new TextDBException("failed to write plan object", e);
-        } finally {
-            try {
-                oos.close();
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
+            // Converting the JSON String to a JSON Node to minimize space usage and to check validity of JSON string
+            JsonParser jsonParser = new JsonParser();
+            JsonObject jsonObject = jsonParser.parse(logicalPlanJson).getAsJsonObject();
+
+            // Writing to the logical JSON string to a file
+            FileWriter fileWriter = new FileWriter(filePath);
+            fileWriter.write(jsonObject.toString());
+            fileWriter.flush();
+            fileWriter.close();
+        }
+        catch (JsonParseException e) {
+            throw new TextDBException("Logical Plan JSON is invalid", e);
+        }
+        catch (IOException e) {
+            throw new TextDBException("Failed to write the Logical Plan JSON", e);
         }
     }
 
```

Comment:
`deletePlanObject` -> `deletePlanJson` ?
Since there are the `writePlanJson` and `readPlanJson` functions.